pkg/gpu/testdata/cudasample.c

```diff
@@ -27,24 +27,30 @@ cudaError_t cudaStreamSynchronize(cudaStream_t stream) {
     return 0;
 }
 
+cudaError_t cudaSetDevice(int device) {
+    return 0;
+}
+
 int main(int argc, char **argv) {
     cudaStream_t stream = 30;
 
-    if (argc < 3) {
-        fprintf(stderr, "Usage: %s <wait-to-start-sec> <wait-to-end-sec>\n", argv[0]);
+    if (argc < 4) {
+        fprintf(stderr, "Usage: %s <wait-to-start-sec> <wait-to-end-sec> <device-index>\n", argv[0]);
```

Comment:
This test binary should be called directly by the code in `samplebins.go` which provides all the arguments, I prefer not to add defaults here as we shouldn't be running this directly and it can also mask failures in the calling code (e.g., we add an argument here but forget to add it in the testing caller code).